route/v1/types.go

```diff
@@ -282,3 +282,19 @@ const (
 	//          Note that this will not match acme.test only *.acme.test
 	WildcardPolicySubdomain WildcardPolicyType = "Subdomain"
 )
+
+// Route Annotations
+const (
+	// AllowNonDNSCompliantHostAnnotation indicates that the host name in a route
+	// configuration is not required to follow strict DNS compliance.
+	// Unless the annotation is set to true, the route host name must have
+	// at least two labels, with each label no more than 63 characters from the set of
+	// alphanumeric characters, '-' or '.', and must start and end with an alphanumeric
+	// character. A trailing dot is allowed. The total host name length must be no more
+	// than 253 characters.
+	// When the annotation is set to true, the host name must pass a smaller set of
+	// requirements, e.g. character set as described above, and total host name
+	// length must be no more than 253 characters.
+	// NOTE: This may validate routes that cannot be admitted and will not function.
```

Comment:
Please tell me why I would want to do this in the doc here.

Reply:
Unfortunately, we don't know why anyone would want to add a host name that might produce a route that wouldn't be admitted. With perhaps an overabundance of caution, this bypass is added only to keep things working as they were before, for someone who relied on this behavior. Please let me know if you need more information.

Reply:
> Unfortunately, we don't know why anyone would want to add a host name that might produce a route that wouldn't be admitted. With perhaps an overabundance of caution, this bypass is added only to keep things working as they were before, for someone who relied on this behavior. Please let me know if you need more information.

you need to place this information into the field doc so that a user knows that they should basically never do this.

Reply:
You can place your hypothetical scenario here so they know.

Reply:
Okay, does my update look reasonable?